### ISSUE TITLE
Exporter ensightgold and createSubmesh.

### ISSUE THREAD

**Issue by vincentchabannes** [feelpp/feelpp#649]
The ensightgold exporter generate a wrong geo file when the mesh exported is obtained from a 3d mesh by extracting a set of faces. This bug appears only in // and for the 3d case with FACES extraction.
It works without problem in 3d with ELEMENTS and EDGES extraction and also for all cases in 2d (ELEMENTS,FACES,EDGES). The format ensight don't have any issue.
For reproduce this : 
```
    auto mesh = loadMesh(_mesh=new Mesh<Simplex<3> >);
    auto submesh = createSubmesh( mesh, boundaryfaces(mesh) );
    auto e = exporter( _mesh=submesh );
    e->addRegions();
    e->save();
```
I searched (without success) if some datas in // mesh was missing. All seems ok and it's similar to the other cases.

**Comment by aancel** [comment]
I'll work on it

**Comment by aancel** [comment]
@vincentchabannes Commit 11673697c6b5990401e807f0481d37568689de7c should fix the issue with your sample code. You can close the issue if everything is ok on your side.
FYI, the problem was caused by bad part ids. They should start at 1 and were 0 on the sample code, so I juste shifted them, but kept the part name at the correct marker id (for display purpose). It is however strange that nobody previously encountered this issue ...

**Comment by vincentchabannes** [comment]
Thanks, it works!
However, the field pid ( ```with e->addRegions() ``` ) is not available in paraview, but it seems exported  and it appears in the properties view in paraview.

**Comment by aancel** [comment]
Apparently, the pid is contained in the data indeed, but there seem to be a problem for the value on the boundaries.
I'll keep looking, seems that there is an export problem with the faces.

**Comment by aancel** [comment]
@vincentchabannes An update on this side: Apparently the pid field can be displayed correctly in EnSight. I'll keep look to find what is the problem with ParaView.

**Comment by prudhomm** [comment]
I consider it closed. Reopen  if it is not

**Comment by vincentchabannes** [comment]
Still a bug with e->addRegions() (probably P0d functions) when load into Paraview

**Comment by aancel** [comment]
@vincentchabannes I am still investigating this issue, and why the add regions does not work correctly.
Can you confirm that with `auto submesh = createSubmesh( mesh, boundaryfaces(mesh) );` I will only end up with faces ?
Do you know when the `exporter.ensightgold.save-face` is actually used ? It was from the previous verison of the exporter and according to the description I might be misuing it in the current version.
(It should also be renamed, if it does what is told in the corresponding description)

**Comment by vincentchabannes** [comment]
no, you manipulate elements with submesh (if mesh is 3d with realdim=3, else submesh is 2d with realdim=3 and elements are triangle. You can also save the face of submesh with your option save-face but in this example the faces (say topological faces) are edges.
In fact, you have exactly the same structure between a 2d mesh and this one obtain with submesh. Just the realdim change but normally there is no impact with an exporter which work always with a realdim=3.

**Comment by prudhomm** [comment]
how about generating a surfacic mesh in 3D and exporting it to ensightgold ? does that work with addRegions ?
